### PR TITLE
#498 Fix for throws low level exception

### PR DESCRIPTION
**Pull request by larsroettig**
see #498

### CODE REVIEW

File: app/code/Magento/InventoryIndexer/Model/GetStockItemData.php [hunk 1]
```diff
@@ -50,6 +51,13 @@ public function execute(string $sku, int $stockId)
             ->from($stockItemTableName, [IndexStructure::QUANTITY, IndexStructure::IS_SALABLE])
             ->where(IndexStructure::SKU . ' = ?', $sku);
 
-        return $connection->fetchRow($select) ?: null;
+        try {
+            return $connection->fetchRow($select) ?: null;
+        } catch (\Exception $e) {
+            throw new LocalizedException(__(
+                'Product with sku "%sku" is not assigned to stock with id "%stock"',
```
Comment (naydav) [inline app/code/Magento/InventoryIndexer/Model/GetStockItemData.php]: If we catch `\Exception` (all of possible exceptions) we need to provide more general error message
(BUT NOT Something went wrong)

File: app/code/Magento/InventoryIndexer/Model/GetStockItemData.php [hunk 2]
```diff
@@ -50,6 +51,13 @@ public function execute(string $sku, int $stockId)
             ->from($stockItemTableName, [IndexStructure::QUANTITY, IndexStructure::IS_SALABLE])
             ->where(IndexStructure::SKU . ' = ?', $sku);
 
-        return $connection->fetchRow($select) ?: null;
+        try {
+            return $connection->fetchRow($select) ?: null;
```
Comment (naydav) [inline app/code/Magento/InventoryIndexer/Model/GetStockItemData.php]: Need to cover this functionality with integration test

**Review by naydav — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryIndexer/Model/GetStockItemData.php (hunk 1 above)
Comment (p-bystritsky) [reply app/code/Magento/InventoryIndexer/Model/GetStockItemData.php]: @naydav done.

File: app/code/Magento/InventoryIndexer/Model/GetStockItemData.php (hunk 2 above)
Comment (p-bystritsky) [reply app/code/Magento/InventoryIndexer/Model/GetStockItemData.php]: @naydav done.

**Review by naydav — APPROVED** [review]

**Pull request merged**